psh/tools/common.py

```diff
@@ -27,7 +27,8 @@
 SEPARATOR_PATTERN = SEPARATOR_PATTERN_BASE + r'+?'
 # acceptable patterns: white spaces (wss) + CC, CC + wss, wss, nothing
 OPTIONAL_SEPARATOR_PATTERN = SEPARATOR_PATTERN_BASE + r'*?'
-CHARS = list(set(string.printable) - set(string.whitespace) - set('/'))
+# all printable chars excluding whitespaces and SH syntax
+CHARS = list(set(string.printable) - set(string.whitespace) - set('/<>;|&'))
```

Comment:
I would add also:
- `()` (subshells?)
- `{}` (control blocks, variable substitiution)
- `[]` (test syntax) 
- `$` (variables)
- `!` (previous commands)
- `*` (to avoid globbing with existing files)
- `\` (beginning of escape sequence so next chars might be treated differently)

bash man page mentions the following special chars:
<img width="544" alt="image" src="https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/3939983/f6f7d94f-446f-4453-831c-7afa9a5d1718">

Reply:
Oh that's right, thanks!

Reply:
now that I think - we should also omit `'`, `"` and backtick (`).

idea: maybe instead of excluding special chars we should just explicitly include some chars instead of thinking what we should exclude?

Reply:
Missed my last comment before the merge?

Reply:
Yes, sorry for that I am now considering change in the next PR